lib/src/executable.dart

```diff
@@ -212,33 +172,44 @@ String _writeSourceMap(ExecutableOptions options, SingleMapping sourceMap) {
 }
 
 /// Runs an interactive SassScript shell.
-_repl() async {
+_repl(ExecutableOptions options) async {
   var repl = new Repl(prompt: '>> ');
-  var environment = new Environment();
+  var variables = <String, internal.Value>{};
   await for (String line in repl.runAsync()) {
     try {
-      _runLine(line, environment);
-    } on SassFormatException catch (e) {
-      print(e);
-    } on SassRuntimeException catch (e) {
-      print(e);
+      Expression expression;
+      VariableDeclaration declaration;
+      try {
+        declaration = new VariableDeclaration.parse(line);
+        expression = declaration.expression;
+      } on SassFormatException {
+        expression = new Expression.parse(line);
+      }
+      var result = evaluateExpression(expression, variables: variables);
+      if (declaration != null) {
+        variables[declaration.name] = result;
+      }
+      print(result);
+    } on SassException catch (error, stackTrace) {
+      var highlighted = error.span.highlight();
+      var arrows = highlighted.split('\n').last.trimRight();
+      var buffer = new StringBuffer();
+      if (options.color) {
+        int start = arrows.length - arrows.trimLeft().length;
+        buffer.write("\u001b[1F"); // move to start of input line
+        buffer.write("\u001b[${start + 3}C"); // move to start of error
+        buffer.write("\u001b[31m"); // set color to red
+        buffer.write(line.substring(start, arrows.length)); // write bad input
```

Comment:
It looks like this line crashes when the syntax error points after the final character of the string. For example, the input `foo(` will cause it to crash.